substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1499,6 +1496,17 @@ impl<T: Config> Pallet<T> {
         pricing_policy: &pallet_tfgrid_types::PricingPolicy<T::AccountId>,
         amount: BalanceOf<T>,
     ) -> DispatchResultWithPostInfo {
+        log::info!(
+            "Distributing cultivation rewards for contract {:?} with amount {:?}",
+            contract.contract_id,
+            amount,
+        );
+
+        // If the amount is zero, return
+        if amount == BalanceOf::<T>::saturated_from(0 as u128) {
+            return Ok(().into());
```

Comment:
Other point
Why do we handle this case inside `_distribute_cultivation_rewards()`
Could be interesting to be more explicit upward like this
~~~rust
        let billable_amount = twin_balance.min(contract_lock.amount_locked);

        if billable_amount != BalanceOf::<T>::saturated_from(0 as u128) {
            // Distribute cultivation rewards
            match Self::_distribute_cultivation_rewards(
                &contract,
                &pricing_policy,
                billable_amount,
            ) { ... }    
        }
   ~~~

Reply:
because we need to wrap an entire code block in an If-statement, I prefer the cleaner early return in the function.